File: backend/src/models/event.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Event struct {
	ID           uint           `json:"id" gorm:"primaryKey"`
	OrganizerID  uint           `json:"organizer_id" gorm:"not null"`
	Organizer    User           `json:"organizer" gorm:"foreignKey:OrganizerID"`
	Type         string         `json:"type" gorm:"not null;size:20;default:'event';check:type IN ('game','event','training')"` // ADD THIS LINE
	Title        string         `json:"title" gorm:"not null;size:255"`
	Description  string         `json:"description" gorm:"type:text"`
	Sport        string         `json:"sport" gorm:"size:100"`
	StartAt      time.Time      `json:"start_at" gorm:"type:timestamptz;not null"`
	EndAt        *time.Time     `json:"end_at" gorm:"type:timestamptz"`
	Capacity     *int           `json:"capacity"`
	LocationName string         `json:"location_name" gorm:"size:255"`
	Latitude     *float64       `json:"latitude" gorm:"not null"`
	Longitude    *float64       `json:"longitude" gorm:"not null"`
	Status       string         `json:"status" gorm:"not null;size:20;default:'upcoming';check:status IN ('upcoming','active','complete','cancelled')"`
	CreatedAt    time.Time      `json:"created_at"`
	UpdatedAt    time.Time      `json:"updated_at"`
	DeletedAt    gorm.DeletedAt `json:"deleted_at" gorm:"index"`
}

File: frontend/src/types/event.ts
export interface Event {
  id: string;
  organizer_id: string;
  type: EventType;
  title: string;
  description: string;
  sport: string;
  start_at: Date;
  end_at?: Date;
  location_name: string;
  latitude: number;
  longitude: number;
  capacity?: number;
  participants: number;
  status: EventStatus;
  created_at: Date;
  updated_at: Date;
}

export interface EventWithOrganizer extends Event {
  organizer_name: string;
  organizer_username: string;
  organizer_avatar?: string;
  is_organizer: boolean;
  is_participant: boolean;
}

export interface CreateEventData {
  type: EventType;
  title: string;
  description: string;
  sport: string;
  start_at: Date;
  end_at?: Date;
  location_name: string;
  latitude: number;
  longitude: number;
  capacity?: number;
}

export interface UpdateEventData {
  type?: EventType;
  title?: string;
  description?: string;
  sport?: string;
  start_at?: Date;
  end_at?: Date;
  location_name?: string;
  latitude?: number;
  longitude?: number;
  capacity?: number;
}

export type EventType = "game" | "event" | "training";

export type EventStatus = "upcoming" | "active" | "complete" | "cancelled";

export interface EventFilters {
  sport?: string;
  type?: EventType;
  location?: string;
  scope?: "all" | "following";
  lat?: number;
  lng?: number;
  radius_km?: number;
  status?: string;
  start_after?: string;
  start_before?: string;
  min_capacity?: number;
  max_capacity?: number;
  limit?: number;
  offset?: number;
}

export interface EventParticipant {
  id: string;
  event_id: string;
  user_id: string;
  role: string;
  joined_at: Date;
  username: string;
  name: string;
  avatar?: string;
}
